codePost_lib.py

```diff
@@ -0,0 +1,695 @@
+#!/usr/bin/env python3
+##########################################################################
+# codePost submission utils
+#
+# DATE:    2019-02-12
+# AUTHOR:  codePost (team@codepost.io)
+#
+##########################################################################
+
+import sys as _sys
+import requests as _requests
+
+try:
+    # Python 2
+    from urllib import quote as _urlquote
+except ImportError:
+    # Python 3
+    from urllib.parse import quote as _urlquote
+
+BASE_URL = 'https://api.codepost.io'
+
+##########################################################################
+
+
+class _color:
+    RED = '\033[91m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+    END = '\033[0m'
+
+
+_TERM_INFO = "{END}[{BOLD}INFO{END}]{END}".format(**_color.__dict__)
+_TERM_ERROR = "{END}[{BOLD}{RED}ERROR{END}]{END}".format(**_color.__dict__)
+
+
+def _print_info(msg):
+    print("{tag} {msg}".format(tag=_TERM_INFO, msg=msg), file=_sys.stderr)
+
+
+DEFAULT_MODE = {
```

Comment:
Do we really want the default behavior of the script to be overwrite everything? I would have thought the maximally cautious approach (all flags set to `False`) would be best.

Reply:
Good point. And no, this was a bogus mode, with copy-pasted `True` for every line. This default mode is essentially OVERWRITE. I guess I should have defined each of the modes before pushing the changes, but I wanted to get another pair of eyes on the logic of the code before doing that—in case the individual criteria might change.